Mathlib/Order/CompleteLattice/SetLike.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2024 Sven Manthe. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sven Manthe
+-/
+import Mathlib.Order.CompleteSublattice
+
+/-!
+# SetLike for CompleteSublattice (Set X)
+
+This file defines a `SetLike` instance for `CompleteSublattice (Set X)`, but does not register it
+for typeclass inference to avoid conflicts.
+-/
+
+namespace CompleteSublattice
+
+variable {X : Type*} {L : CompleteSublattice (Set X)}
+
+/-- membership is inherited from `Set X` -/
+@[simps] def setLikeCompleteSublattice : SetLike L X where
+  coe T := T
+  coe_injective' := Subtype.val_injective
+
+section
+
+attribute [local instance] setLikeCompleteSublattice
+
+variable {S T : L} {s : Set L} {I : Sort*} {f : I → L} {x : X}
```

Comment:
I'm a bit surprised by the naming of variables: I'd expect `s : S` and not `S: s`.